### DISCUSSION TITLE
Windows Setup

### DISCUSSION THREAD

**Discussion by NexuaTCN** [uhppoted/uhppoted#28] [Q&A] [answered]
Hello! Awesome project! So, I'm trying to connect to my controller on windows, but seem to be having some trouble. When I try to use ">uhppote-cli get-devices" it originally did nothing, but after i configured the device in the conf file, I'm receiving an error "Failed to write to UDP socket: A socket operation was attempted to an unreachable network.". I can't seem to figure out what I'm doing wrong. Thanks in advance for any help!

**Comment by uhppoted** [comment, marked as answer]
Hi,

Hmm, best guess is that you are running multiple networks (e.g. WiFi and one or two Ethernet LANs) and _uhppote-cli_ is not using the correct one - it defaults to binding to either `0.0.0.0` or to the first LAN it finds (if there are multiple interfaces). As a start, try removing all the configuration information from _uhppoted.conf_ (to get to a known state) and then:

- Use `ipconfig` to list all your networks
```
ipconfig /all
```
You should get something like this:
```
Windows IP Configuration

   Host Name . . . . . . . . . . . . : MY-COMPUTER
   Primary Dns Suffix  . . . . . . . : example.com
   Node Type . . . . . . . . . . . . : Hybrid
   IP Routing Enabled. . . . . . . . : No
   WINS Proxy Enabled. . . . . . . . : No
   DNS Suffix Search List. . . . . . : example.com

Ethernet adapter Ethernet:

   Connection-specific DNS Suffix  . : example.com
   Description . . . . . . . . . . . : Realtek PCIe GbE Family Controller
   Physical Address. . . . . . . . . : 00-11-22-33-44-55
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   IPv4 Address. . . . . . . . . . . : 192.168.1.2(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Lease Obtained. . . . . . . . . . : Friday, April 15, 2023 10:28:18 AM
   Lease Expires . . . . . . . . . . : Saturday, April 16, 2023 10:28:18 AM
   Default Gateway . . . . . . . . . : 192.168.1.1
   DHCP Server . . . . . . . . . . . : 192.168.1.1
   DNS Servers . . . . . . . . . . . : 192.168.1.1
...
```

- You want to use the Ethernet adapter for the network that your access controller is on and then bind specifically to that adapter, e.g.:

```
uhppote-cli --debug --bind 192.168.1.2 --broadcast 192.168.1.255 get-devices
```

- You can also try explicitly binding to the INADDR_ANY address (which may or may not work depending on how your network is configured):

```
uhppote-cli --debug --bind 0.0.0.0 --broadcast 255.255.255.255 get-devices
```

There are a couple of other potential reasons but they're a bit less likely:
- The access controller is on a different network segment to the host computer
- Your access controller may have very out of date firmware (pre v6.56). This is unlikely unless it's a very old controller - some of the less expensive controllers sold on AliExpress are _old stock_ but even so the oldest version I've heard of is v6.62.
- Your network may be configured to block UDP broadcast
- You're not running a UHPPOTE controller (_sorry, I do have to ask - it has happened :-)_)

**Comment by NexuaTCN** [comment]
You're the best! It was using a virtual adapter, still beating myself up haha. Thank for your help, got it working now. Really appreciate your work!

**Comment by uhppoted** [comment]
Ok, that is one I haven't run into before :-) ... glad you're up and running!

I'm guessing you've probably already done this, but if not you can set the default _bind_, _broadcast_ and _listen_ addresses in the _uhppoted.conf_ file, e.g.:
```
# SYSTEM
bind.address = 192.168.1.100
broadcast.address = 192.168.1.255
listen.address = 192.168.1.100:60001

```
